internal/test/providers/aws/aws.go

```diff
@@ -0,0 +1,404 @@
+package aws
+
+import (
+	"encoding/json"
+	"fmt"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	awssession "github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/aws/aws-sdk-go/service/iam"
+	mapi "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	"github.com/openshift/windows-machine-config-bootstrapper/internal/test/clusterinfo"
+	"k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	awsprovider "sigs.k8s.io/cluster-api-provider-aws/pkg/apis/awsprovider/v1beta1"
+)
+
+const (
+	infraIDTagKeyPrefix = "kubernetes.io/cluster/"
+	infraIDTagValue     = "owned"
+	windowsLabel        = "machine.openshift.io/os-id"
+	// instanceType is the AWS specific instance type to create the VM with
+	instanceType = "m5a.large"
+)
+
+type awsProvider struct {
+	// imageID is the AMI image-id to be used for creating Virtual Machine
+	imageID string
+	// instanceType is the flavor of VM to be used
+	instanceType string
+	// A client for IAM.
+	iam *iam.IAM
+	// A client for EC2. to query Windows AMI images
+	ec2 ec2iface.EC2API
+	// openShiftClient is the client of the existing OpenShift cluster.
+	openShiftClient *clusterinfo.OpenShift
+	// region in which the Machine needs to be created
+	region string
+	// sshKeyPair is the key pair associated with the Windows VM
+	sshKeyPair string
+}
+
+// newSession uses AWS credentials to create and returns a session for interacting with EC2.
+func newSession(credentialPath, credentialAccountID, region string) (*awssession.Session, error) {
+	if _, err := os.Stat(credentialPath); os.IsNotExist(err) {
+		return nil, fmt.Errorf("failed to find AWS credentials from path '%v'", credentialPath)
```

Comment:
You should be checking for errors in general. permission errors, etc.
Checking specifically if not exist is useful for if an entity can create the file in question if its missing, in this case this is not possible

Reply:
basically what we are doing here is that , we are checking if the file exists and in case it does not exist it'll throw a `*PathError`. With `os.IsNotExist()` takes the error and returns a boolean indicating whether the error is known to report that a file or directory does not exist. Our intention here is to find if the credentials file is mounted. I am not sure about a different way to implement this.

Reply:
So what happens if you get a different error? You will just proceed and fail later?

Reply:
if there is some other error that causes the requests to AWS to error out. It'll fail later.